### ISSUE TITLE
Remove deprecated twisted.web.google

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5768]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5768|
|Type|task|
|Created|2012-07-11 11:04:39Z|
|Branch|https://github.com/twisted/twisted/tree/remove-google-5768|

`twisted.web.google` was deprecated in Twisted 11.1 ([#5209](https://github.com/twisted/twisted/issues/5209)) and can be removed (inc. the sample).

Attachments:

* [google-5678.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c35/c354adc107dd9dae8b0f48c2bdfbac9a7de0b0f4/00709e1ed8ccc6b34505e52cd40d5db632c5f3f6.patch) (5624 bytes) - added by thijs on 2012-07-11 11:11:29Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5768 5768
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_remove_google_5768 branches/remove-google-5768
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1342004679000000 1342004679000000
changetime__1342346752000000 1342346752000000
version__None None
owner__spiv spiv
cc__thijs cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to spiv|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|spiv commented|
|-|-|

(In [34792]) Branch to remove-google-5768

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|spiv commented|
|-|-|

Tests pass locally, and it's hard to object to deleting code!  I don't have a single nitpick to make.

Assuming buildbot likes this I'll submit it.  Thank you!

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|spiv set status to `closed`|
|-|-|

(In [34796]) Merge remove-google-5768: Remove the deprecated twisted.web.google module.

Author: thijs
Reviewer: spiv
Fixes: [#5768](https://github.com/twisted/twisted/issues/5768)

Removes twisted.web.google, which was deprecated in Twisted 11.1, and
associated documentation.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph commented|
|-|-|

Hello eager deprecators (especially Thijs):

In this case I don't object to the removal, because the code we're talking about here can arguably be said to violate Google's terms of service and we should never have shipped it in the first place.  (You should be using the google [custom search API](https://developers.google.com/custom-search/v1/overview) if you want to automate things, which this module never supported.)

However, tickets like this should be worded so as to explain that the code should be removed _for a reason_ (such as "maybe it violates Google's terms of service").  The fact that code _can_ be removed does not necessarily mean that it _should_ be.  Deprecated code should be removed for a specific reason, when it becomes an unreasonable maintenance burden, or if its presence is for some reason confusing or problematic.

Generally, we should hang on to deprecated functionality unless it starts to cause us problems and there is a reason to remove it; doing otherwise is by definition breaking people's programs for no reason.

So, again, no problem with this specific ticket (there _is_ a great reason to remove this code, it just wasn't clearly described), but I am concerned over the general pattern of removing things as soon as we possibly can according to policy.  In the future, let's start with a description of why we want to remove it.

Thanks,

-glyph
